src/cpp/core/system/PosixSystem.cpp

```diff
@@ -2472,10 +2472,6 @@ Error permanentlyDropPriv(const std::string& newUsername)
    if (error)
       return error;
 
-   // before changing the process user, ensure it becomes the new
-   // owner of any file logs so we can ensure that we can keep writing to them
-   core::log::refreshAllLogDestinations(core::log::RefreshParams{ user });
```

Comment:
For 2 here, I believe that would only affect a fork without an exec, like with daemonize.  Or it could affect log statements done after the fork but before the exec. That doesn't seem like a problem to me since it seems like we always call exec and I could not find any log statements in that window between fork and exec.

Reply:
There are several possible failure points between a `fork` and `exec` that can generating logging calls. See this entire codeblock where we have just forked, need to perform more setup before exec, but all of those syscalls can potentially fail:

https://github.com/rstudio/rstudio/blob/8aaf9b559c59dce270f0cc6c6d23516b841f0962/src/cpp/core/system/PosixChildProcess.cpp#L566-L607

Reply:
Ah, yes, I see that now. I put back the call to refreshAllLogDestinations - just without the user. Still works so I think we are all set now.  Thanks for your help!

Reply:
This is better but still leaves potential for the following scenario:

1. Server runs a child process as the same user - does not change user (see for example `rserver-pam` or `server-session-reaper`.
2. Forks
3. Forked child has an error
4. Error is not logged properly since log destinations were not refreshed (this may not actually be an issue - this can be fairly easily tested by forcefully injecting an error condition in code when spawning a child process like `rserver-pam`).

We really should call `refreshAllLogDestinations` without user change data in any forked child. Otherwise it is possible we can miss log statements.

`temporarilyDropPriv` can continue to call `refreshAllLogDestinations` as it is not used within a forked child.

Reply:
I thought that's what I did in the code... I just replaced the old refreshAllLogDestinations(...user...) with refreshAllLogDestinations().  For some reason the new code showed up below this comment thread though so maybe you didn't see it? Or am I missing some other cases where we need this call?

Reply:
Right, that's not sufficient because that is being called in the case where we are running `permanentlyDropPriv`. There are cases where `permanentlyDropPriv` is skipped entirely (when not changing users), though we still are a forked child that should invoke `refreshAllLogDestinations`. There's quite a few different spots in the code where this happens.  A grep of `::fork` and `fork()` should show you where.

Reply:
Ok, that makes sense. I see a few of those places now. Given where we are in the release cycle, I think we need to address this later so I'll open up a new issue.

Reply:
Excellent. Thanks!